### ISSUE TITLE
Issue with restAPI room creation and Broadcast Presence 

### ISSUE THREAD

**Issue by jdclark77** [igniterealtime/openfire-restAPI-plugin#49]
Hi,
I am updating our OpenFire servers to use the latest versions of both server and plugin from a much older version (4.11).  We use the restAPI from our .Net code base by calling the end points using ,Net HttpClient (this is just for info and to let you know I am not using one the provided clients).  When I create the room, I as part of the JSON it sends 
... "broadcastPresenceRoles": ["moderator", "participant", "visitor"] .. 
however when I check the room created on the server room admin pages, these values are not checked.  Also, I can see the XMPP stanzas in my web client (using F12 dev tools) and the server is not sending the stanzas (probably because it's not set).  The server was set in a two node cluster until I disabled it to debug this issue and once in prod, it will federate with another, just more back ground info.  I looked at the issues but I'm not seeing anything about this.  Are there any know issues with the rest client and broadcast presence?  I know the create room parameters for broadcast presence changed after plugin version 1.3.0 ( we still use old one in prod).  Also ant other issues I may hit with cluster or federation?
Openfire server 4.6.1
restAPI 1.4.0
Hazelcast 2.5.0

sample full JSON
{
    "roomName": "room-1",
    "naturalName": "room-1",
    "description": "Global chat room",
    "maxUsers": "0",
    "publicRoom": "true",
    "persistent": "true",
    "registrationEnabled": "false",
    "canAnyoneDiscoverJID": "false",
    "canOccupantsChangeSubject": "false",
    "canOccupantsInvite": "false",
    "canChangeNickname": "true",
    "logEnabled": "true",
    "loginRestrictedToNickname": "false",
    "membersOnly": "true",
    "moderated": "false",
    "broadcastPresenceRoles": [
	    "moderator",
	    "participant",
	    "visitor"
    ],
    "owners": [],
    "admins": [],
    "members": []
    "outcasts": []
}

Thanks ahead for any help with this.

**Comment by tyrex1975** [comment]
Having the exact same problem here.

Interestingly, broadcastPresenceRoles is populated as per requirements when you attempt to GET the room after creating it.

Whereas when you create the room in the admin client (with the Broadcast Presence values ticked), the REST API cannot retrieve the room and this error log is produced in Openfire.

```
at   javax.servlet.http.HttpServlet.service(HttpServlet.java:790)   ~[javax.servlet-api-3.1.0.jar:3.1.0]
--
at   org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1452)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:791)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1626)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.jivesoftware.admin.PluginFilter.doFilter(PluginFilter.java:226)   ~[xmppserver-4.6.1.jar:4.6.1]
at   org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.jivesoftware.admin.AuthCheckFilter.doFilter(AuthCheckFilter.java:234)   ~[xmppserver-4.6.1.jar:4.6.1]
at   org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   com.opensymphony.sitemesh.webapp.SiteMeshFilter.obtainContent(SiteMeshFilter.java:129)   ~[sitemesh-2.4.2.jar:?]
at   com.opensymphony.sitemesh.webapp.SiteMeshFilter.doFilter(SiteMeshFilter.java:77)   ~[sitemesh-2.4.2.jar:?]
at   org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:201)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602)   ~[jetty-security-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)   ~[jetty-servlet-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:191)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.Server.handle(Server.java:516)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)   ~[jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)   [jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)   [jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)   [jetty-server-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)   [jetty-io-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)   [jetty-io-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)   [jetty-io-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)   [jetty-util-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)   [jetty-util-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)   [jetty-util-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)   [jetty-util-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)   [jetty-util-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)   [jetty-util-9.4.35.v20201120.jar:9.4.35.v20201120]
at   org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)   [jetty-util-9.4.35.v20201120.jar:9.4.35.v20201120]
at   java.lang.Thread.run(Unknown Source) [?:1.8.0_202]
Caused   by: java.lang.ClassCastException: org.jivesoftware.openfire.muc.MUCRole$Role   cannot be cast to java.lang.String
at   org.codehaus.jackson.map.ser.std.IndexedStringListSerializer.serializeContents(IndexedStringListSerializer.java:91)   ~[?:?]
...   67 more
```

So it would seem the REST API not updating the internal room.broadcastModerator, room.broadcastParticipant and room.broadcastVisitor parameters.

**Comment by ty-ac** [comment]
I was having the same issue so I decided to take a look at the code and I find out it has been broken since OpenFire 4.6.1.

In the git commit `62ab9734dfb1f08795e8ac5b47e4a5d55ee8b51a` ( aka “`OF-2168: MUC presence broadcast supression based on role`”), there was a change in the signature of the method `getRolesToBroadcastPresence()` and `setRolesToBroadcastPresence()` of the class `MUCRoom` and since the REST API plugin is based on older version of the plugins API, it has been broken.

Right now the maven repository of OpenFire is still referring to the version 4.6.0, so it’s not possible to easily rebuild the REST API plugin. Hopefully they’ll update the maven repository soon, this way a new version of the plugin may be released as well.

**Comment by patelnishantk** [comment]
Facing the same issue with v4.6.3. I hope openfire team can provide some fix soon.

**Comment by dashrathgoswami** [comment]
We also facing the same issue since the update to the new version 4.6.3 also in 4.6.2, it seems the API version is no longer compatible with the new version and breaking down everything, it is working fine with version 4.6.0

More info [Here](https://discourse.igniterealtime.org/t/error-500-retrieving-rooms-info-with-rest-api-plugin/89851)

**Comment by gywreb** [comment]
Hi , any update on this issue ?

**Comment by guusdk** [comment]
This should be fixed in e98bc36c874328e829e8572061eb5f01c6189562
